src/contexts/RefreshContext.tsx

```diff
@@ -1,52 +1,34 @@
-import React, { useState, useEffect, useRef } from 'react'
+import React, { useState, useEffect } from 'react'
+import useIsWindowVisibleRef from 'hooks/useIsWindowVisibleRef'
 
 const FAST_INTERVAL = 10000
 const SLOW_INTERVAL = 60000
 
 const RefreshContext = React.createContext({ slow: 0, fast: 0 })
 
-// Check if the tab is active in the user browser
-const useIsBrowserTabActive = () => {
-  const isBrowserTabActiveRef = useRef(true)
-
-  useEffect(() => {
-    const onVisibilityChange = () => {
-      isBrowserTabActiveRef.current = !document.hidden
-    }
-
-    window.addEventListener('visibilitychange', onVisibilityChange)
-
-    return () => {
-      window.removeEventListener('visibilitychange', onVisibilityChange)
-    }
-  }, [])
-
-  return isBrowserTabActiveRef
-}
-
 // This context maintain 2 counters that can be used as a dependencies on other hooks to force a periodic refresh
 const RefreshContextProvider = ({ children }) => {
   const [slow, setSlow] = useState(0)
   const [fast, setFast] = useState(0)
-  const isBrowserTabActiveRef = useIsBrowserTabActive()
+  const isWindowVisibleRef = useIsWindowVisibleRef()
 
   useEffect(() => {
     const interval = setInterval(async () => {
-      if (isBrowserTabActiveRef.current) {
+      if (isWindowVisibleRef.current) {
```

Comment:
I don't think this works. When you wrap it in a `useRef` it will no longer trigger effects when the window becomes visible again.

Reply:
Yes, in refresh context I just extract that to utils since we have more than one usage. It is used like that to not trigger useRefresh every time we have an event. Otherwise every tab switch will send requests.

Reply:
Okay, so in this case instead of creating another hook that copies the functionality just return a ref as well.

In `useIsWindowVisible` the return can look like this

```
return { isWindowVisible, isWindowVisibleRef }
```

Reply:
That should stay separated since hook that contains useState triggers rendering even if we only use ref object

Reply:
In this case I think it is okay as it is not an event that fires frequently.

Reply:
Then I am closing the PR for now